src/main/java/apoc/bolt/Bolt.java

```diff
@@ -0,0 +1,160 @@
+        }
+    }
+
+    @Procedure()
+    @Description("apoc.bolt.execute(url-or-key, statement, params, config) - access to other databases via bolt for read")
+    public Stream<RowResult> execute(@Name("url") String url, @Name("statement") String statement, @Name(value = "params", defaultValue = "{}") Map<String, Object> params, @Name(value = "config", defaultValue = "{}") Map<String, Object> config) throws URISyntaxException {
+        Map<String, Object> configuration =  new HashMap<>(config);
+        configuration.remove("virtual");
+        return load(url, statement, params, configuration);
+    }
+
+    private StatementResult runStatement(@Name("statement") String statement, Session session, Map<String, Object> finalParams) {
+        return session.writeTransaction((Transaction tx) -> tx.run(statement, finalParams));
+    }
+
+    private Stream<RowResult> getRowResultStream(boolean virtual, Session session, Map<String, Object> params, String statement) {
+        return runStatement(statement, session, params).list().stream()
+                .map(record -> record.fields())
+                .map(fields -> {
+                    Map<String, Object> map = new HashMap<>();
+                    for (int i = 0; i < fields.size(); i++) {
+                        String key = fields.get(i).key();
+                        Object value = fields.get(i).value().asObject();
+                        if (value instanceof Path) map.putAll(toPath(value, virtual, key));
+                        else if (value instanceof Node) map.putAll(toNode(value, key, virtual));
+                        else if (value instanceof Relationship) map.putAll(toRelationship(value, map, key, virtual));
+                        else map.put(key, value);
+                    }
+                    return map;
+                })
+                .map(resultMap -> new RowResult(resultMap));
+    }
+
+    private Map<String, Object> toNode(Object value, String key, boolean virtual) {
+        Value internalValue = ((InternalEntity) value).asValue();
+        Node node = internalValue.asNode();
+        if (virtual) {
+            VirtualNode virtualNode = new VirtualNode(node.id(), db);
+            node.labels().forEach(l -> virtualNode.addLabel(Label.label(l)));
+            node.asMap().entrySet().iterator().forEachRemaining(i -> virtualNode.setProperty(i.getKey(), i.getValue()));
+            return Util.map(key, virtualNode);
+        } else
+            return Util.map(key, Util.map("entityType", internalValue.type().name(), "labels", node.labels(), "id", node.id(), "properties", node.asMap()));
+    }
+
+    private Map<String, Object> toRelationship(Object value, Map<String, Object> map, String key, boolean virtual) {
+        Value internalValue = ((InternalEntity) value).asValue();
+        Relationship relationship = internalValue.asRelationship();
+        if (virtual) {
+            VirtualNode start = new VirtualNode(relationship.startNodeId(), db);
```

Comment:
we should use a "cache" of virtual nodes that we construct elsewhere and use "lazy" virtual node implementation that we just construct with an id and the cache and  it would redirect on access and only create the vnode if it doesn't exist yet.